src/storage/operations/mv.rs

```diff
@@ -0,0 +1,180 @@
+            let entry_norm = entry_path.trim_start_matches('/');
+            if meta.mode() == EntryMode::DIR && entry_norm == src_norm {
+                continue;
+            }
+
+            let relative_path = get_root_relative_path(entry_path, src_path);
+            let new_dest_path = build_remote_path(dest_path, &relative_path);
+
+            if meta.mode() == EntryMode::DIR {
+                self.ensure_directory(&new_dest_path).await?;
+            } else {
+                self.stream_move(entry_path, &new_dest_path).await?;
+                self.operator.delete(entry_path).await?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Stream move a single file with progress reporting.
+    async fn stream_move(&self, src_path: &str, dest_path: &str) -> opendal::Result<()> {
+        let metadata = self.operator.stat(src_path).await?;
+        let file_size = metadata.content_length();
+
+        let mut writer = self.operator.writer(dest_path).await?;
+        let mut total_bytes = 0u64;
+        let mut offset = 0u64;
+
+        let reporter = ConsoleProgressReporter::new(
+            format!("Moving {src_path}"),
+            Some(file_size),
+            DEFAULT_CHUNK_SIZE as u64,
+        );
+
+        loop {
+            let chunk_size = std::cmp::min(DEFAULT_CHUNK_SIZE as u64, file_size - offset);
+
+            let data = self
+                .operator
+                .read_with(src_path)
+                .range(offset..offset + chunk_size)
+                .await?;
+            let data_len = data.len();
+            if data_len == 0 {
+                break;
+            }
+
+            writer.write(data).await?;
+            total_bytes += data_len as u64;
+            offset += chunk_size;
+
+            reporter.maybe_report(total_bytes);
+        }
+
+        writer.close().await?;
+        println!("\n✅ Moved: {src_path} → {dest_path} ({total_bytes} bytes)");
+
+        Ok(())
+    }
+}
+
+impl Mover for OpenDalMover {
+    async fn mover(&self, src_path: &str, dest_path: &str) -> Result<()> {
+        let src_stat = self.operator.stat(src_path).await.ok();
+        let src_is_dir = self.is_directory(src_path).await;
+        ensure!(
+            src_stat.is_some() || src_is_dir,
+            InvalidPathSnafu {
+                path: src_path.to_string()
+            }
+        );
+
+        if src_is_dir {
+            let target_root = if self.is_directory(dest_path).await {
+                let base_name = basename(src_path);
+                let target_root = build_remote_path(dest_path, &base_name);
+                self.ensure_directory(&target_root).await?;
+                target_root
+            } else {
+                self.ensure_directory(dest_path).await?;
+                dest_path.to_string()
+            };
+
+            self.move_file_recursive(src_path, &target_root).await?;
+            Ok(())
+        } else {
+            let dest_is_dir_hint = dest_path.ends_with('/');
```

Comment:
instead of erroring when dest_path ends with / but directory doesn’t exist, create it and treat as directory.